pkg/endpoint/endpoint.go

```diff
@@ -360,14 +381,125 @@ func (e *Endpoint) WaitForProxyCompletions() error {
 	return nil
 }
 
+// runK8sCiliumEndpointSync starts a controller that syncronises the endpoint
+// to the corresponding k8s CiliumEndpoint CRD
+// ep.Mutex must be help for the setup portion of this function, but not the
+// controller.
+func (e *Endpoint) runK8sCiliumEndpointSync() {
+	elog := e.getLogger()
+	endpointID := e.ID
+
+	controllerName := fmt.Sprintf("sync-to-k8s-ciliumendpoint (%v)", endpointID)
+	epName := fmt.Sprintf("%v-cep-%v", node.GetName(), endpointID)
+
+	if !k8s.IsEnabled() {
+		elog.WithField("name", controllerName).Debug("Not starting controller because k8s is disabled")
+		return
+	}
+
+	ciliumClient, err := getCiliumClient()
```

Comment:
This will create a client for every `runK8sCiliumEndpointSync` invocation. Can't you share a client for all endpoints?

Reply:
I should be able to. I'll switch it over.